Orange/projection/radviz.py

```diff
@@ -1,47 +1,38 @@
 import numpy as np
 
-from Orange.data import Domain
-
-
-def radviz(data, attrs, points=None):
-    x = data.transform(domain=Domain(attrs)).X
-    mask = ~np.isnan(x).any(axis=1)
-    x = x[mask]
-
-    n = len(x)
-    if not n:
-        return None, None, mask
-    x = normalize(x)
-
-    r_x = np.zeros(n)
-    r_y = np.zeros(n)
-
-    m = x.shape[1]
-    if points is not None:
-        s = points
-    else:
-        s = np.array([(np.cos(t), np.sin(t))
-                      for t in [2.0 * np.pi * (i / float(m))
-                                for i in range(m)]])
-    for i in range(n):
-        row = x[i]
-        row_ = np.repeat(np.expand_dims(row, axis=1), 2, axis=1)
-        with np.errstate(divide='ignore', invalid='ignore'):
-            a = (s * row_).sum(axis=0)
-            b = row.sum()
-            y = np.divide(a, b, out=np.zeros_like(a), where=b != 0)
-        r_x[i] = y[0]
-        r_y[i] = y[1]
-
-    return np.stack((r_x, r_y), axis=1), s, mask
-
-
-def normalize(x):
-    """
-    MinMax normalization to fit a matrix in the space [0,1] by column.
-    """
-    a = x.min(axis=0)
-    b = x.max(axis=0)
-    diff = b - a
-    diff[diff == 0] = 1
-    return (x - a[np.newaxis, :]) / diff[np.newaxis, :]
+from Orange.preprocess.preprocess import RemoveNaNRows, Continuize, Normalize
+from Orange.projection import LinearProjector
+
+__all__ = ["RadViz"]
+
+
+class RadViz(LinearProjector):
```

Comment:
There is the `self.projection = None` in the parent widget's clear().

Reply:
Yes, but I still think this should be a class attribute (or even a method?!) because all instances use the same method. I'll look into this again, later today. And even if I decide I believe it is necessary, I promise not to bother you with this if it would require any substantial changes in the code.

Reply:
We gave up. :)